core/src/main/clojure/xtdb/sql.clj

```diff
@@ -2666,7 +2665,7 @@
                     {doc ~(into {} (map (juxt keyword identity)) known-cols)}]
           [:order-by [[_iid] [_valid_from]]
            [:scan {:table ~table,
-                   :for-valid-time [:in ~valid-from ~valid-to]}
+                   :for-valid-time [:in ~(or valid-from :now) ~valid-to]}
```

Comment:
`nil` is now an expression so I need to be explicit here and also in `xtql/plan.clj` below.